src/ContractionTreeToGraph.jl

```diff
@@ -0,0 +1,120 @@
+
+"""The main object here is `g' a NamedGraph which represents a graphical version of a contraction sequence.
+It's vertices describe a partition between the leaves of the sequence (will be labelled with an n = 1 or n = 3 element tuple, where each element of the tuple describes the leaves in one of those partition)
+n = 1 implies the vertex is actually a leaf.
+Edges connect vertices which are child/ parent and also define a bi-partition"""
+
+
+"""Function to take a sequence (returned by ITensorNetworks.contraction_sequence) and construct a graph g which represents it (see above)"""
+function create_contraction_tree(contract_sequence)
+    #Collect all the leaves
+    leaves = collect(Leaves(contract_sequence))
+    #Build empty graph
+    g = NamedGraph()
+    #Construct the left branch
+    spawn_child_branch(contract_sequence[1], g, leaves)
+    #Construct the right branch
+    spawn_child_branch(contract_sequence[2], g, leaves)
+
+    #Now we have the vertices we need to figure out the edges
+    for v in vertices(g)
+        #Only add edges from a parent (which defines a tripartition and thus has length 3) to its children
+        if(length(v) == 3)
+        #Work out which vertices it connects to
+            concat1, concat2, concat3 =[v[1]..., v[2]...], [v[2]..., v[3]...], [v[1]..., v[3]...]
+            for vn in setdiff(vertices(g), [v])
+                vn_set = [Set(vni) for vni in vn]
+                if(Set(concat1) ∈ vn_set || Set(concat2) ∈ vn_set || Set(concat3) ∈ vn_set)
+                    add_edge!(g, v => vn)
+                end
+            end
+        end
+    end
+
+
+    return g
+end
+
+"""Given a contraction sequence which is a subsequence of some larger sequence which is being built on current_g and has leaves `leaves`
+Spawn `contract sequence' as a vertex on `current_g' and continue on with its children """
+function spawn_child_branch(contract_sequence, current_g, leaves)
+    #Check if sequence is at leaf point
+    if(isa(contract_sequence, Array))
+        #If not it  will spawn two children
+        group1 = collect(Leaves(contract_sequence[1]))
+        group2 = collect(Leaves(contract_sequence[2]))
+        remaining_verts = setdiff(leaves, vcat(group1, group2))
+        add_vertex!(current_g, (group1, group2, remaining_verts))
+        spawn_child_branch(contract_sequence[1], current_g, leaves)
+        spawn_child_branch(contract_sequence[2], current_g, leaves)
+    else
+        #If it is it is just a vertex
+        add_vertex!(current_g, ([contract_sequence], setdiff(leaves, [contract_sequence])))
+    end
+end
+
+"""Utility functions for the graphical representation of a contraction sequence
+Perhaps it should be a specific object or type of tree?!"""
+
+"""Determine if a node is a leaf"""
+function leaf_node(g::AbstractGraph, v)
+    return length(neighbors(g, v)) == 1
+end
+
+"""Determine if an edge involves a leaf (at src or dst)"""
+function leaf_edge(g::AbstractGraph, e)
```

Comment:
A better name would be `is_leaf_edge`.

I've been putting functions like this that are generic to all graph types in the `src/Graphs/abstractgraph.jl` file of `NamedGraphs.jl`. Maybe for now we could move these to a `src/Graphs/abstractgraph.jl` of `ITensorNetworks.jl` and then move them to `NamedGraphs.jl` later.